src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs

```diff
@@ -21,14 +22,38 @@ internal static partial class AnalyzerOptionsExtensions
         private static readonly ImmutableHashSet<OutputKind> s_defaultOutputKinds =
             ImmutableHashSet.CreateRange(Enum.GetValues(typeof(OutputKind)).Cast<OutputKind>());
 
+        private static bool TryGetSyntaxTreeForOption(ISymbol symbol, [NotNullWhen(returnValue: true)] out SyntaxTree? tree)
+        {
+            switch (symbol)
```

Comment:
💡 Switching over the symbol _kind_ will be more efficient than interface type checks.

Reply:
Done, thanks!